src/build/patches.ts

```diff
@@ -265,15 +298,24 @@ function handleMethod(child: Node): Partial<Method> {
     }
   }
 
-  const signature: Method["signature"] = [
-    {
-      param: params,
-      ...(typeNode
-        ? handleTyped(typeNode)
-        : { type: string(child.properties?.returns) }),
-    },
-  ];
-  return { name, signature };
+  const signature:
+    | DeepPartial<Signature>[]
+    | Record<string, DeepPartial<Signature>> = child.properties?.overrideType
+    ? {
+        "0": {
```

Comment:
This certainly is the key but I don't think we should do it this way, What if we only want to override the second signature but leave the first one as-is?

What about something like:

```kdl
method name signatureIndex=0 type=undefined
```

And then we return object as signature list instead of an array if `signatureIndex` exists?